starport/cmd/network_chain_publish.go

```diff
@@ -47,16 +53,27 @@ func NewNetworkChainPublish() *cobra.Command {
 
 func networkChainPublishHandler(cmd *cobra.Command, args []string) error {
 	var (
-		source        = args[0]
-		tag, _        = cmd.Flags().GetString(flagTag)
-		branch, _     = cmd.Flags().GetString(flagBranch)
-		hash, _       = cmd.Flags().GetString(flagHash)
-		genesisURL, _ = cmd.Flags().GetString(flagGenesis)
-		chainID, _    = cmd.Flags().GetString(flagChainID)
-		campaign, _   = cmd.Flags().GetUint64(flagCampaign)
-		noCheck, _    = cmd.Flags().GetBool(flagNoCheck)
+		source            = args[0]
+		tag, _            = cmd.Flags().GetString(flagTag)
+		branch, _         = cmd.Flags().GetString(flagBranch)
+		hash, _           = cmd.Flags().GetString(flagHash)
+		genesisURL, _     = cmd.Flags().GetString(flagGenesis)
+		chainID, _        = cmd.Flags().GetString(flagChainID)
+		campaign, _       = cmd.Flags().GetUint64(flagCampaign)
+		noCheck, _        = cmd.Flags().GetBool(flagNoCheck)
+		mainnet, _        = cmd.Flags().GetBool(flagMainnet)
+		totalSupplyStr, _ = cmd.Flags().GetString(flagTotalSupply)
 	)
 
+	if mainnet && campaign == 0 && totalSupplyStr == "" {
+		return errors.New("you should use the --mainnet flag along with the --campaign or --total-supply flag")
+	}
+
+	totalSupply, err := sdk.ParseCoinsNormalized(totalSupplyStr)
```

Comment:
Campaign and total-supply flags cannot be set together. We need to add a check for this and return an error if both provided together.

Also:

We should probably move this line below, into the `else` stmt of `if campaign != 0 {`.

And same for `publishOptions = append(publishOptions, network.WithTotalSupply(totalSupply))`, it should be added into the  `else` stmt of `if campaign != 0 {`.